pkg/operator/ceph/object/spec.go

```diff
@@ -912,3 +922,70 @@ func renderProbe(cfg rgwProbeConfig) (string, error) {
 
 	return writer.String(), nil
 }
+
+func (c *clusterConfig) addDNSNamesToRGWServer() (string, error) {
+	if (c.store.Spec.Hosting == nil) || len(c.store.Spec.Hosting.DNSNames) <= 0 {
+		return "", nil
+	}
+	if !c.clusterInfo.CephVersion.IsAtLeastReef() {
+		return "", errors.New("rgw dns names are supported from ceph v18 onwards")
+	}
+
+	// add default RGW service name to dns names
+	dnsNames := c.store.Spec.Hosting.DNSNames
+	dnsNames = append(dnsNames, domainNameOfService(c.store))
+
+	// add custom endpoints from zone spec if exists
+	if c.store.Spec.Zone.Name != "" {
+		zone, err := c.context.RookClientset.CephV1().CephObjectZones(c.store.Namespace).Get(c.clusterInfo.Context, c.store.Spec.Zone.Name, metav1.GetOptions{})
+		if err != nil {
+			return "", err
+		}
+		for _, endpoint := range zone.Spec.CustomEndpoints {
+			domainName, err := URLParse(endpoint)
+			if err != nil {
+				return "", err
+			}
+			if domainName != "" {
+				dnsNames = append(dnsNames, domainName)
+			}
+		}
+	}
+
+	// remove duplicate dns names
+	checkDuplicate := make(map[string]bool)
+	removeDuplicateDNSNames := []string{}
+	for _, dnsName := range dnsNames {
+		if _, ok := checkDuplicate[dnsName]; !ok {
+			checkDuplicate[dnsName] = true
+			removeDuplicateDNSNames = append(removeDuplicateDNSNames, dnsName)
+		}
+	}
+
+	err := validateDNS1123(removeDuplicateDNSNames)
+	if err != nil {
+		return "", err
+	}
+
+	return cephconfig.NewFlag("rgw dns name", strings.Join(removeDuplicateDNSNames, ",")), nil
+}
+
+// Validate DNSName if it is codified by RFC-1123
+func validateDNS1123(dnsNames []string) error {
+	var err []string
+	for _, i := range dnsNames {
+		err = validation.IsDNS1123Subdomain(i)
+		if err != nil {
+			return errors.Errorf(strings.Join(err, " "), "invalid DNS-1123 subdomain %q", i)
+		}
+	}
+	return nil
+}
+
+func URLParse(endpoint string) (string, error) {
```

Comment:
This function's intent is to extract the hostname from an endpoint. Please name it accordingly.

```suggestion
func GetHostname(endpoint string) (string, error) {
```